pkg/proxy/iptables/proxier.go

```diff
@@ -980,6 +986,26 @@ func (proxier *Proxier) syncProxyRules() {
 		proxier.endpointChainsNumber += len(proxier.endpointsMap[svcName])
 	}
 
+	// If set localhostNodePorts to "false"(route_localnet=0), We should generate iptables rules that
+	// disable NodePort services to be accessed via localhost. Otherwise. it doesn't work and causes
+	// the kernel to log warnings if anyone tries.
+	if !proxier.localhostNodePorts && proxier.iptables.Protocol() == utiliptables.ProtocolIPv4 {
+		proxier.natRules.Write(
```

Comment:
Unless I am missing something, this is wrong.  This rule doesn't just disallow NodePorts via localhost - it disallows ALL traffic from localhost.

We jump to KUBE-NODEPORTS at the end of KUBE-SERVICES.  So we will end up in KUBE-NODEPORTS for all sorts of traffic.

I think you either need to:

a) Change the jump to KUBE-NODEPORTS to use `! -d 127.0.0.1/8` (and IPv6 equiv).  There's a loop of nodeAddresses which we need to decide what to do if that includes localhost - which opinion wins?  I think if should filter out localhost and log an error.
 
b) Change each NodePort match rule to use `! -d 127.0.0.1/8` (and IPv6 equiv).

Reply:
> Change the jump to KUBE-NODEPORTS to use ! -d 127.0.0.1/8 (and IPv6 equiv).

yeah, I think this is correct (and solves the `kubeMarkDropChain` problem too; we don't need to explicitly drop the packets. We can just refrain from acting on them.

Reply:
Thanks for the review!

Please review the latest changes of this PR,  I'm not sure these changes are appropriate and I would appreciate if you could give some suggestions.